tests/functional/u/unused/unused_private_member.py

```diff
@@ -201,3 +201,19 @@ def __inner_4():  # [unused-private-member]
 
         fn_to_return = __inner_1 if flag else __inner_3(__inner_2)
         return fn_to_return
+
+
+# Test cases for crash reported in #4755
```

Comment:
That line is more or less obvious from the Github link below. Better to write a short description here. Something like
```py
# https://github.com/PyCQA/pylint/issues/4755
# Nested attributes shouldn't cause crash
```

(I would also reverse the order and put the link first.)

Reply:
Fixed this, also added a few overlooked testcases and fixed docs for some of the other test cases i've worked on.

I realise that this might bloat up this PR a little, and I'm sorry for that.

Hope it's okay :pray:

Reply:
👌🏻